src/client/Microsoft.Identity.Client/Internal/Requests/UsernamePasswordRequest.cs

```diff
@@ -45,7 +50,19 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             await ResolveAuthorityAsync().ConfigureAwait(false);
             await UpdateUsernameAsync().ConfigureAwait(false);
 
-            MsalTokenResponse msalTokenResponse = await GetTokenResponseAsync(cancellationToken).ConfigureAwait(false);
+            MsalTokenResponse msalTokenResponse = null;
+                
+            try
+            {
+                msalTokenResponse = await GetTokenResponseAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch (JsonException ex)
+            {
+                if (_requestParameters.AuthorityInfo.AuthorityType == AuthorityType.B2C)
```

Comment:
if it isn't a B2C exception, you should not swallow exception, you need to re-throw it. When rethrowing, make sure to use `throw;` instead of `throw ex;` because `throw;` will preserve the stack trace. This is a blocking comment.

Reply:
I would catch a JsonException on line 54, then here throw a more general message, something like "There was an error parsing the response from the token endpoint. Verify that your app is configured correctly. If this is a B2C app, one possible cause is acquiring a token for Microsoft Graph, which is not supported."

Because this Json parsing error can happen in other cases, so we're just future proofing this way.

Reply:
Since this error happens because instead of Json, AAD returns an HTML error page, we could also log that response into ErrorPii I guess.

Reply:
Thanks for the explanation @pmaytak. Wasn't able to repro this scenario and hence thought catching everything for B2C scenario is better. This makes more sense. Will update the error message and catch just JsonException. CC: @bgavrilMS

Reply:
> Since this error happens because instead of Json, AAD returns an HTML error page, we could also log that response into ErrorPii I guess, but that would require more work since that code is inside token client classes, so would be out of scope for this work item.

In fact, thinking about this more, that error page will probably be hard to read in the logs anyway, so disregard that comment.

Reply:
Ok, was not planning to log it as it is HTML page.